weaviate/backup/backup.py

```diff
@@ -0,0 +1,351 @@
+"""
+Backup class definition.
+"""
+from time import sleep
+from typing import Union, List
+from weaviate.exceptions import (
+    UnexpectedStatusCodeException,
+    RequestsConnectionError,
+    BackupFailedException,
+)
+from weaviate.util import _capitalize_first_letter
+from weaviate.connect import Connection
+
+
+STORAGE_NAMES = [
```

Comment:
I thought about enums, but I do not want to expose it to the users. It is used only for argument validation.
But I can add it anyway.

Reply:
Enums are going to add more complexity than I we think we need, since I do not intend to expose it to the users I think a `set` would work good as well. Also you cannot just check if a value is in `Enum` without workarounds, which is annoying.